project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module

```diff
@@ -97,24 +97,32 @@ function c4m_content_form_node_form_alter(&$form, &$form_state, $form_id) {
   // Use destination parameter when set.
   $params = drupal_get_query_parameters();
   if (empty($params['destination'])) {
-    // Use javascript history.go(-1) when destination is not set.
-    $form['actions']['cancel'] = array(
-      '#type' => 'button',
-      '#value' => t('Cancel'),
-      '#attributes' => array('onClick' => 'history.go(-1); return true;'),
-      '#weight' => 14,
-    );
-  }
-  else {
-    $form['actions']['cancel'] = array(
-      '#type' => 'submit',
-      '#value' => t('Cancel'),
-      '#submit' => array('c4m_content_cancel_button_callback'),
-      '#limit_validation_errors' => array(),
-      '#weight' => 14,
+    $path_tokens = explode('/', rtrim($form['#action'], '/'));
```

Comment:
@anvmn 
there is a lot of logic in here.
please add examples in an elaborated comment.